civis/futures.py

```diff
@@ -214,15 +221,30 @@ def __init__(self, job_id, run_id,
                  polling_interval=None,
                  client=None,
                  poll_on_creation=True):
-        if client is None:
-            client = APIClient(resources='all')
-        super().__init__(client.scripts.get_containers_runs,
-                         [int(job_id), int(run_id)],
-                         polling_interval=polling_interval,
-                         client=client,
-                         poll_on_creation=poll_on_creation)
+        Future.__init__(self)
+        self.client = client or APIClient(resources='all')
+        self.poller_args = [int(job_id), run_id]
+        self._polling_interval = polling_interval
+        self._poll_on_creation = poll_on_creation
         self._max_n_retries = max_n_retries
 
+        if run_id is not None:
+            self.poller_args[1] = int(run_id)
+            self._init_polling_pubnubbing()
+
+    def _init_polling_pubnubbing(self):
+        super().__init__(self.client.scripts.get_containers_runs,
+                         self.poller_args,
+                         polling_interval=self._polling_interval,
+                         client=self.client,
+                         poll_on_creation=self._poll_on_creation)
+
+    def _start_container(self):
```

Comment:
Fixed.

Reply:
Yeah this creating all sorts of problems since this method also inits the parent class. I have had to add an extra switch which makes the code too confusing. I think this change is too DRY.